pymdownx/snippets.py

```diff
@@ -107,22 +108,25 @@ def parse_snippets(self, lines, file_name=None):
                     continue
 
                 snippet = os.path.join(self.base_path, path)
-                if snippet and os.path.exists(snippet):
-                    if snippet in self.seen:
-                        # This is in the stack and we don't want an infinite loop!
-                        continue
-                    if file_name:
-                        # Track this file.
-                        self.seen.add(file_name)
-                    try:
-                        with codecs.open(snippet, 'r', encoding=self.encoding) as f:
-                            new_lines.extend(
-                                [space + l2 for l2 in self.parse_snippets([l.rstrip('\r\n') for l in f], snippet)]
-                            )
-                    except Exception:  # pragma: no cover
-                        pass
-                    if file_name:
-                        self.seen.remove(file_name)
+                if snippet:
+                    if os.path.exists(snippet):
+                        if snippet in self.seen:
+                            # This is in the stack and we don't want an infinite loop!
+                            continue
+                        if file_name:
+                            # Track this file.
+                            self.seen.add(file_name)
+                        try:
+                            with codecs.open(snippet, 'r', encoding=self.encoding) as f:
+                                new_lines.extend(
+                                    [space + l2 for l2 in self.parse_snippets([l.rstrip('\r\n') for l in f], snippet)]
+                                )
+                        except Exception:  # pragma: no cover
+                            pass
+                        if file_name:
+                            self.seen.remove(file_name)
+                    elif not os.path.exists(snippet) and self.check_paths: # pragma: no cover

```

Comment:
Okay, this works too, but we don't need to check if snippet exists here as we already did it in the original, just change this to `elif self.check_paths`  and remove `# pragma: no cover` as we want to exercise this path, and ensure it is covered.  Please add a test that covers this.

Reply:
Fixed and tested.